### PR TITLE
E2E & Node E2E: Add exec util in framework

### PR DESCRIPTION
**Pull request by Random-Liu**
For #29081.
Based on #29092 and #29494.

For first commit is a squashed commit of all old commits.
**The last 2 commits are new.**

This PR added exec util in framework, and moved `privileged.go` and `kubelet_etc_hosts` into `common` directory.

@vishh @timstclair 
/cc @kubernetes/sig-node

### CODE REVIEW

File: test/e2e/framework/exec_util.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+		Command:   cmd,
+		Stdin:     stdin != nil,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       tty,
+	}, api.ParameterCodec)
+
+	err = execute("POST", req.URL(), config, stdin, &stdout, &stderr, tty)
+	Expect(err).NotTo(HaveOccurred(), "post request failed")
+	Logf("Exec stderr: %q", stderr.String())
```
Comment (timstclair) [inline test/e2e/framework/exec_util.go]: nit: only log if stderr is not empty

Comment (Random-Liu) [inline test/e2e/framework/exec_util.go]: ACK.

**Comment by timstclair** [comment]
LGTM

**Comment by Random-Liu** [comment]
Removed almost all dependency commits, they are not required for this PR.

Apply LGTM based on https://github.com/kubernetes/kubernetes/pull/29536#issuecomment-235117694.

**Pull request merged**